.github/workflows/publish_docker.yml

```diff
@@ -0,0 +1,41 @@
+name: publish docker
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish_docker.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish_docker.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - tag: "7.0" #Debian 11
+            file: Dockerfile
+          - tag: 7.0-jammy # Ubuntu 22.04
+            file: Dockerfile
+          - tag: 7.0-alpine # Alpine 3.17
+            file: Dockerfile.alpine
+          - tag: 7.0-alpine3.16
+            file: Dockerfile.alpine
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: setup docker buildx
+        uses: docker/setup-buildx-action@v2
+      - name: publish
+        shell: pwsh
+        env:
+          TAG: ${{ matrix.tag }}
+          DOCKERFILE: ${{ matrix.file }}
+        run: |
+          docker buildx build --build-arg "TAG=$env:TAG" -f $env:DOCKERFILE -o out .
```

Comment:
I wanted to initially try docker compose and not need to create the workflow like this but they don't support the `output` flag

Reply:
the `-o out` saves the published executable into the out folder on the github runner

Reply:
I thought it could be easier to upload them outside of the dockerfile and in the workflow when we get to that point